sw/airborne/modules/guidance/gvf_parametric/gvf_parametric.cpp

```diff
@@ -345,14 +361,85 @@ bool gvf_parametric_2D_trefoil_wp(uint8_t wp, float w1, float w2, float ratio, f
 { 
   gvf_parametric_trajectory.p_parametric[7] = wp;
   gvf_parametric_plen_wps = 1;
-
+  #if defined(FIXEDWING_FIRMWARE)
   gvf_parametric_2D_trefoil_XY(waypoints[wp].x, waypoints[wp].y, w1, w2, ratio, r, alpha);
+  #elif defined(ROVER_FIRMWARE)
+  gvf_parametric_2D_trefoil_XY(WaypointX(wp), WaypointY(wp), w1, w2, ratio, r, alpha);
+  #endif
   return true;
 }
 
+// 2D CUBIC BEZIER CURVE
+bool gvf_parametric_2D_bezier_XY(void)
+{
+	gvf_parametric_trajectory.type = BEZIER_2D;
+	float fx, fy, fxd, fyd, fxdd, fydd;
+	gvf_parametric_2d_bezier_splines_info(gvf_bezier_2D, &fx, &fy, &fxd, &fyd, &fxdd, &fydd);
+	gvf_parametric_control_2D(gvf_parametric_2d_bezier_par.kx, gvf_parametric_2d_bezier_par.ky, fx, fy, fxd, fyd, fxdd, fydd);
+	return true;
+}
+
+// TODO: Improve scalability (pass an array of wp) ??
+bool gvf_parametric_2D_bezier_wp(uint8_t wp0, uint8_t wp1, uint8_t wp2, uint8_t wp3, uint8_t wp4, uint8_t wp5, uint8_t wp6, uint8_t wp7, uint8_t wp8, uint8_t wp9,
+				  uint8_t wp10, uint8_t wp11, uint8_t wp12)
```

Comment:
This function is not very convenient to use. I suggest to pass only 2 parameters: the id of the first WP and the number of WP. It means that all WPs used for bezier curves must be defined in the correct order in the flight plan, but this is acceptable constraint (already used for other navigation routines like https://github.com/paparazzi/paparazzi/blob/master/sw/airborne/modules/nav/nav_survey_poly_rotorcraft.h#L45).

Reply:
Since the number of points is fixed by the number of Bézier segments, I've only passed the first point of the spline and then the user shall define in the airframe file the number of segments. Thanks for the help, using this new way improves scalability a lot.